goosebit/api/v1/software/routes.py

```diff
@@ -69,11 +71,15 @@ async def post_update(_: Request, file: UploadFile | None = File(None), url: str
         # local file
         artifacts_dir = Path(config.artifacts_dir)
         file_path = artifacts_dir.joinpath(file.filename)
+        tmp_file_path = artifacts_dir.joinpath("tmp", ("".join(random.choices(string.ascii_lowercase, k=12)) + ".tmp"))
+        await tmp_file_path.parent.mkdir(parents=True, exist_ok=True)
 
-        async with aiofiles.tempfile.NamedTemporaryFile("w+b") as f:
+        async with await open_file(tmp_file_path, "w+b") as f:
             await f.write(await file.read())
-            absolute = await file_path.absolute()
-            software = await create_software_update(absolute.as_uri(), Path(f.name))
+        absolute = await file_path.absolute()
+        tmp_absolute = await tmp_file_path.absolute()
+        software = await create_software_update(absolute.as_uri(), tmp_absolute)
+        await tmp_file_path.unlink(missing_ok=True)
```

Comment:
Move temp file deletion in a finally block to ensure that it always gets deleted.